### ISSUE TITLE
OneCloud fails on Android 9

### ISSUE THREAD

**Issue by mvysny** [mvysny/photocloud-frame-slideshow#103]
```
Fatal Exception: java.lang.NoClassDefFoundError: Failed resolution of: Lorg/apache/http/impl/client/DefaultHttpClient;
       at com.microsoft.services.msa.LiveAuthClient.<init>(LiveAuthClient.java:253)
       at com.microsoft.services.msa.LiveAuthClient.<init>(LiveAuthClient.java:318)
       at com.onedrive.sdk.authentication.MSAAuthenticator.init(MSAAuthenticator.java:143)
       at com.onedrive.sdk.authentication.DisambiguationAuthenticator.init(DisambiguationAuthenticator.java:129)
       at com.onedrive.sdk.extensions.OneDriveClient$Builder.loginAndBuildClient(OneDriveClient.java:163)
       at com.onedrive.sdk.extensions.OneDriveClient$Builder$1.run(OneDriveClient.java:147)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:764)
```

**Comment by mvysny** [comment]
Reported as bug in OneCloud client library: https://github.com/OneDrive/onedrive-sdk-android/issues/150

**Comment by mvysny** [comment]
Workarounds:

* Adding `compile 'org.apache.httpcomponents:httpclient:4.5.8'` leads to `java.lang.IncompatibleClassChangeError: Class 'org.apache.http.conn.ssl.SSLSocketFactory' does not implement interface 'org.apache.http.conn.scheme.SchemeSocketFactory' in call to 'java.net.Socket org.apache.http.conn.scheme.SchemeSocketFactory.createSocket(org.apache.http.params.HttpParams)' (declaration of 'org.apache.http.impl.conn.DefaultClientConnectionOperator'` as in https://stackoverflow.com/questions/54414287/using-org-apache-http-legacy-in-android-9/54414655

**Comment by mvysny** [comment]
Fixed in PhotoCloud 1.13.8
